ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -33,9 +69,17 @@ cpdef traceback_to_frames(traceback, max_nframes):
     while tb is not None:
         if nframes < max_nframes:
             frame = tb.tb_frame
-            code = frame.f_code
-            lineno = 0 if frame.f_lineno is None else frame.f_lineno
-            frames.insert(0, (code.co_filename, lineno, code.co_name, _extract_class_name(frame)))
+            # Python 3.11 moved PyFrameObject members to internal C API and cannot be directly accessed
+            IF PY_MAJOR_VERSION > 3 or (PY_MAJOR_VERSION == 3 and PY_MINOR_VERSION >= 11):
+                code = PyFrame_GetCode(<PyFrameObject*> frame)
+                lineno = PyFrame_GetLineNumber(<PyFrameObject*> frame)
+                lineno = 0 if lineno is None else lineno
+                frames.insert(0, ((<object>code).co_filename, lineno, (<object>code).co_name, _extract_class_name(frame)))
+                Py_XDECREF(<PyObject*>code)
```

Comment:
we might want a try/finally ? what happens if we get an error, we don't decref this.

Reply:
Yep, the only "weak" line that I can see here in this regard is the "pure Python" `frames.insert` statement, as this could cause exceptions, but with the guarantees around `code` I think it is unlikely it will throw an exception.